### PR TITLE
Train with just one dataset

### PR DESCRIPTION
**Pull request by GrantPerkins**
Not sure if multiple datasets ever worked, or if it was ever a good idea. Now it is just one dataset.

### CODE REVIEW

**Review by AustinShalit — COMMENTED** [review]
We should also update the UI  to only permit a single selection and change the database relationship to many-to-one to enforce this behavior.

**Comment by GrantPerkins** [comment]
@technoAl is on the case!

**Comment by GrantPerkins** [comment]
Many-to-one now enforced, with great difficulty. Radio buttons!

File: packages/server/src/store/index.ts [hunk 1]
```diff
@@ -1,10 +1,13 @@
 import {
   Association,
+  BelongsToGetAssociationMixin,
```
Comment (AustinShalit) [inline packages/server/src/store/index.ts]: Please see lint error on this line - https://github.com/wpilibsuite/Axon/pull/213/checks?check_run_id=2888541211

File: containers/mobilenet/dataset/scripts/tar_to_record.py [hunk 2]
```diff
@@ -25,15 +25,18 @@ def main(dataset_paths, percent_eval, directory):
...
+#         for i in dataset_paths:
+        shutil.copy(dataset_paths, join(EXTRACT_PATH, 'data.tar'))
+
+#         try:
+#             for i in dataset_paths:
+#                 shutil.copy(i, join(EXTRACT_PATH, 'data.tar'))
+#         except:
+#             print('unable to retrieve a dataset tar file:')
+#             sys.exit(1)
+#         for dataset in dataset_paths:
```
Comment (AustinShalit) [inline containers/mobilenet/dataset/scripts/tar_to_record.py]: Clean up please.

**Review by AustinShalit — CHANGES_REQUESTED** [review]

File: packages/client/src/components/project/input/Buttons.tsx [hunk 3]
```diff
@@ -48,7 +48,9 @@ export function StartButton(props: { id: string; selected: GetProjectData_projec
 
   const dockerState = useQuery<GetDockerState>(GET_DOCKER_STATE, { pollInterval: 5000 });
   const datasets = useQuery<GetDatasets, GetDatasets>(GET_DATASETS);
-  const datasetNames = props.selected.map((dataset) => dataset.name);
+
+  let datasetName = "";
+  if (props.selected !== null) datasetName = props.selected.name;
```
Comment (AustinShalit) [inline packages/client/src/components/project/input/Buttons.tsx]: This seems like a weird pattern. Can we move the `props.selected !== null` into the 2 if statements on lines 65 and 75 and remove the datasetName variable?

File: packages/client/src/components/project/input/Datasets.tsx [hunk 4]
```diff
@@ -27,18 +27,22 @@ const useStyles = makeStyles((theme: Theme) =>
...
+    datasetName = props.selected.name;
+  }
 
   if (loading) return <p>LOADING</p>;
   if (error || !data) return <p>ERROR</p>;
 
   /* the function for showing selected datasets inside the input */
   function renderFunc(selected: unknown) {
-    return <div>{datasetNames.join(", ")}</div>;
+    return <div>{datasetName}</div>;
```
Comment (AustinShalit) [inline packages/client/src/components/project/input/Datasets.tsx:45]: Try something like this and remove the code above on lines 34-38. Reach out if it doesn't work. I probably have a question mark in the wrong spot.

```suggestion
    return <div>{props.selected?.name ?? "Dataset"}</div>;
```

File: packages/client/src/components/project/input/Datasets.tsx [hunk 5]
```diff
@@ -27,18 +27,22 @@ const useStyles = makeStyles((theme: Theme) =>
...
+export default function Datasets(props: { id: string; selected: GetProjectData_project_dataset | null }): ReactElement {
   const classes = useStyles();
 
   const { data, loading, error } = useQuery<GetDatasets, GetDatasets>(GET_DATASETS);
-  const datasetNames = props.selected.map((dataset) => dataset.name);
+
+  let datasetName = "Dataset";
+  if (props.selected != null) {
+    datasetName = props.selected.name;
+  }
```
Comment (AustinShalit) [inline packages/client/src/components/project/input/Datasets.tsx]: Try something like this and remove the code above on lines 34-38. Reach out if it doesn't work. I probably have a question mark in the wrong spot.

```suggestion
  const datasetName = props.selected?.name ?? "Dataset";
```

I am also not convinced that we want a default of "dataset". Shouldn't it be null/none?

**Comment by AustinShalit** [comment]
I am leaving for work and don't have time to test, but I want to make sure that we test to see what happens if we try to have a project with no datasets.

File: packages/client/src/components/project/input/Datasets.tsx (hunk 5 above)
Comment (GrantPerkins) [reply packages/client/src/components/project/input/Datasets.tsx]: @technoAl is there still a default?

Comment (technoAl) [reply packages/client/src/components/project/input/Datasets.tsx]: it's the empty string "" It works well

**Review by AustinShalit — APPROVED** [review]

**Pull request merged**